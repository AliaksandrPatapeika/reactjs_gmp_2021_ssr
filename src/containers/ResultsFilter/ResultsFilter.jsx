import './ResultsFilter.less';

import {assign} from 'lodash';
import {useRouter} from 'next/router';
import React from 'react';
import {useDispatch, useSelector} from 'react-redux';

import {setFilter} from '../../actions/movies';
import FilterItem from '../../components/FilterItem';
import {FilterItems} from '../../constants';
import {getQueryString} from '../../utils';

const ResultsFilter = () => {
  const genres = {filterItems: FilterItems};
  const filter = useSelector((state) => state.query.filter);
  const query = useSelector((state) => state.query);
  const dispatch = useDispatch();
  const router = useRouter();
  const {id} = router.query;

  const setFilterHandler = (genre) => {
    // /* for the filtering when the film details page is open */
    if (id) {
      dispatch(setFilter(genre));
    } else {
      router.push({
        pathname: '/search',
        search: getQueryString(assign(query, {filter: genre}))
      });
    }
  };

  return (
    <nav>
      <ul className="resultsFilterContainer">
        {genres.filterItems.map((item) => (
          <FilterItem
            label={item.label}
            key={item.id}
            isActive={item.value === filter}
            setFilter={() => setFilterHandler(item.value)}
          />
        ))}
      </ul>
    </nav>
  );
};

export default ResultsFilter;
