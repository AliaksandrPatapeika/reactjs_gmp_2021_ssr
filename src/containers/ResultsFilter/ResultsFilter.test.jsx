import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import ResultsFilter from './ResultsFilter';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {id: '12'}
  })
}));

describe('ResultsFilter', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      query: {
        filter: ''
      }
    };
    const store = mockStore(initialState);
    const resultsFilter = renderer
      .create(
        <Provider store={store}>
          <ResultsFilter />
        </Provider>
      )
      .toJSON();

    expect(resultsFilter).toMatchSnapshot();
  });
});
