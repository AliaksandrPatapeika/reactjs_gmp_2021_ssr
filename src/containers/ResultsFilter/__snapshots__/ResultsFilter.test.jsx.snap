// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFilter renders correctly 1`] = `
<nav>
  <ul
    className="resultsFilterContainer"
  >
    <li
      className="filterItem active"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex="0"
    >
      ALL
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex="0"
    >
      DOCUMENTARY
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex="0"
    >
      COMEDY
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex="0"
    >
      HORROR
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex="0"
    >
      CRIME
    </li>
  </ul>
</nav>
`;
